Initial Contracts

yonathan9669 opened the pull request:
## Description

This merge contains the initial version of the contracts to be used for the elections creation.

uklok-agent committed:
feat(contract): 👻 Election Registry

- The Election module SmartContract was created.
- The main structure and usability is in place.
- The Interface was also included.
- The deployment was orchestrated to link the 3 contracts and theirs respective roles.
- The Main contract was updated to include the middleware functions to the elections registry.
- Both registries now have the direct reference to the parent.

This ref: #3

uklok-agent committed:
Merge branch 'feat/initial_contracts' into develop

uklok-agent commented:
Approved!!

uklok-agent reviewed (approved):
Several TODO, but it's a good path to track the progress.

uklok-agent committed:
fix(contract): 👻 Solidity Version

- The solidity version of the main contract was updated to prevent CI/CD issues.

This ref: #2

uklok-agent committed:
fix(contract): 👻 Withdraw Function

- The contracts now have a withdrawal function only accessible by the ADMIN_ROLE.
- Some address variables were updated to payable for compatibility.
- The contract versions were assigned.

This ref: #4

uklok-agent committed:
feat(hardhat): 👻 Avalanche Network

- The contracts were deployed on AVAX Fuji.

This ref: #4
